### DISCUSSION TITLE
What is the recommended/idiomatic approach to reusable styles?

### DISCUSSION THREAD

**Discussion by disillusionedowl** [mantinedev/mantine#3837] [Q&A] [answered]
Let's say I have a style that represents some specific red shadow that I want to apply on some (different) components, including some inner elements of those components.

Example of how I would approach this reusable composition in classic HTML/CSS:

```css
.red-shadow {
  box-shadow: 0 0 0 20px red;
}
```

```html
<div class="red-shadow">I have red shadow.</div>
```

Do I add it as another shadow in theme.shadows, or theme.other? Or into Global styles? Or create e.g. `reusable.styles.ts`, import it where needed and merge it together with `cx` (or without) with `mycompXYZ.styles.ts` that I create for each component? Or would it be better to somehow merge styles together directly inside the `mycompXYZ.styles.ts` files and not in the main component files (with `cx`) for better separation of concerns/readability?

**Comment by rtivital** [comment, marked as answer]
I would save this red shadow value in `theme.other` and apply it to components using `createStyles`. All the solutions that you've mentioned will work fine as well.

**Reply by disillusionedowl** [reply]
Thank you @rtivital ! I have ended up doing it like this, leveraging the spread `...` syntax in case I want to add more CSS properties later:

```css
other: {
  myContentBox: {
    boxShadow: '0 0 0 20px red',
    padding: '30px'
  }
}
```
```js
export const useStyles = createStyles( theme => ( {
  root: {
    margin: '0 0 10px 0',
    ...theme.other.myContentBox
  }
} ) );
```

But writing it now I just realized one possible caveat. I don't think I can access the fully constructed `theme` object inside `theme.other`? But I understand it could be the chicken&egg kind of problem. Although `theme.globalStyles` for example seems to have that covered with:

```js
globalStyles: theme => ( { ...... } )
```

Is there any way to solve this caveat?

**Reply by disillusionedowl** [reply]
Nevermind, I have actually ended up doing something similar to this approach https://github.com/mantinedev/mantine/discussions/1205#discussioncomment-3491671 which relies on `cx` and Mantine's [Styles API](https://mantine.dev/styles/styles-api/). I have created various `X.styles.js` files for styles I want to share around and then merge them similarly to this:

```js
const { classes: cssInput, cx } = useInputStyles();
const { classes: cssVariousShared } = useVariousSharedStyles();

<Input
    classNames= { {
        root: cx( cssVariousShared.redShadow, cssInput.root )
    } }
/>
```

Quite verbose, but It is the most versatile solution I found, by far. I can mix in any other classes, including Tailwind, etc. And bonus points for not being global (because "global is bad"). I can just pull in the styles as dependencies when/where needed.
